tools/sigclient/pkg/query/validation.go

```diff
@@ -0,0 +1,664 @@
+			return "", nil, fmt.Errorf("ReadAndValidateQueryFile: Error fetching uniqueKeyCols, uniqueKeyCols: %v file: %v", uniqueKeyCols, filePath)
+		}
+		err = populateTotalMatchedAndRecords(expectedResult, expRes)
+		if err != nil {
+			return "", nil, fmt.Errorf("ReadAndValidateQueryFile: Error populating totalMatched and records, file: %v, err: %v", filePath, err)
+		}
+		if len(expRes.Records) == 0 || len(expRes.Records) > 10 {
+			return "", nil, fmt.Errorf("ReadAndValidateQueryFile: Number of records should be in range 1-10 for logs-query, got: %v file: %v", len(expRes.Records), filePath)
+		}
+		err = ValidateUniqueKeyColsInResult(expRes)
+		if err != nil {
+			return "", nil, fmt.Errorf("ReadAndValidateQueryFile: Error validating UniqueKeyCols, file: %v, err: %v", filePath, err)
+		}
+	}
+
+	return query, expRes, nil
+}
+
+func EvaluateQueryForAPI(dest string, queryReq map[string]interface{}, qid int, expRes *Result) error {
+	query := queryReq["searchText"].(string)
+
+	reqBody, err := json.Marshal(queryReq)
+	if err != nil {
+		return fmt.Errorf("EvaluateQueryForAPI: Error marshaling request, reqBody: %v, err: %v", reqBody, err)
+	}
+
+	url := fmt.Sprintf("http://%s/api/search", dest)
+	req, err := http.NewRequestWithContext(context.TODO(), http.MethodPost, url, bytes.NewReader(reqBody))
+	if err != nil {
+		return fmt.Errorf("EvaluateQueryForAPI: Error creating request, url: %v, reqBody: %v, err: %v", url, reqBody, err)
+	}
+	req.Header.Set("content-type", "application/json")
+
+	sTime := time.Now()
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("EvaluateQueryForAPI: Error sending request, req: %v, err: %v", req, err)
+	}
+	defer resp.Body.Close()
+
+	responseData := make(map[string]interface{})
+	if bodyBytes, err := io.ReadAll(resp.Body); err != nil {
+		return fmt.Errorf("EvaluateQueryForAPI: Error reading response, resp.Body: %v, err: %v", resp.Body, err)
+	} else {
+		if err := json.Unmarshal(bodyBytes, &responseData); err != nil {
+			return fmt.Errorf("EvaluateQueryForAPI: Error unmarshaling bodyBytes: %v, err: %v", string(bodyBytes), err)
+		}
+	}
+
+	queryRes, err := CreateResult(responseData)
+	if err != nil {
+		return fmt.Errorf("EvaluateQueryForAPI: Error creating result, responseData: %v, err: %v", responseData, err)
+	}
+	hits, err := getHits(responseData)
+	if err != nil {
+		return fmt.Errorf("EvaluateQueryForAPI: Error getting hits, responseData: %v, err: %v", responseData, err)
+	}
+	err = populateTotalMatchedAndRecords(hits, queryRes)
+	if err != nil {
+		return fmt.Errorf("EvaluateQueryForAPI: Error populating totalMatched and records, hits: %v, responseData: %v, err: %v", hits, responseData, err)
+	}
+	err = CompareResults(queryRes, expRes, query)
+	if err != nil {
+		return fmt.Errorf("EvaluateQueryForAPI: Failed query: %v, responseData: %v, err: %v", query, responseData, err)
+	}
+
+	log.Infof("EvaluateQueryForAPI: Query %v was succesful. In %+v", query, time.Since(sTime))
```

Comment:
add the time unit, and display the number in humanize.comma format

Reply:
With %v parameter, it would show time with unit.
For example:
```
INFO[0030] EvaluateQueryForWebSocket: Query app_name=Bracecould | fields app_name, batch, http_status, ident was succesful. In 11.285875ms 
INFO[0030] EvaluateQueryForAPI: Query app_name=Bracecould | fields app_name, batch, http_status, ident was succesful. In 8.570875ms 
```
I don't think there is a point in using humanize.Comma since, the query execution be done in ms usually (10-200 ms) and if the value is larger, the %v should handle this by itself.